src/app/clusters/ota-requestor/GenericOTARequestorDriver.cpp

```diff
@@ -64,18 +71,20 @@ void GenericOTARequestorDriver::Init(OTARequestorInterface * requestor, OTAImage
 
     if (mImageProcessor->IsFirstImageRun())
     {
-        SystemLayer().ScheduleLambda([this] {
-            CHIP_ERROR error = mImageProcessor->ConfirmCurrentImage();
-
-            if (error != CHIP_NO_ERROR)
-            {
-                ChipLogError(SoftwareUpdate, "Failed to confirm image: %" CHIP_ERROR_FORMAT, error.Format());
-                mRequestor->Reset();
-                return;
-            }
-
-            mRequestor->NotifyUpdateApplied();
-        });
+        ScheduleDelayedAction(
+            mDelayNotifyUpdateAppliedSeconds,
+            [](System::Layer *, void * context) {
+                CHIP_ERROR error = ToDriver(context)->mImageProcessor->ConfirmCurrentImage();
+                if (error != CHIP_NO_ERROR)
+                {
+                    ChipLogError(SoftwareUpdate, "Failed to confirm image: %" CHIP_ERROR_FORMAT, error.Format());
```

Comment:
If we fail to confirm current image, will we implement rolling back to previous working version?

Reply:
Not for v1. It was discussed that reverting back to a previous good version can be done implicitly by the bootloader.